libparsec/crates/crypto/src/sodium/key_derivation.rs

```diff
@@ -50,18 +50,20 @@ impl KeyDerivation {
 
         let subkey_id = u64::from_le_bytes(*id_low);
         let context = id_high;
+        debug_assert_eq!(context.len(), blake2b::CONTEXTBYTES);
 
-        let mut subkey = GenericArray::default();
-        derive_from_key(&mut subkey, subkey_id, *context, &self.0)
+        let subkey = blake2b::derive_from_key(Size::USIZE, subkey_id, context, &self.0)
             .expect("subkey has always a valid size");
 
-        subkey
+        // generic-array@1.2.0 provide better way to convert from an array but we are stuck by
```

Comment:
```suggestion
    // TODO: replace `from_exact_iter` by `from_array` once generic-array is updated to v1.0+
```

Cheap recycle since this comment is already in the codebase ^^
The important this though is to have a `TODO` marker so we can easily go back to it later on